### PR TITLE
Trace clock settings

### PR DESCRIPTION
**Pull request by ktrzcinx**
Add a few more logs related with clock change, to improve debugging process

### CODE REVIEW

File: src/ipc/handler.c [hunk 1]
```diff
@@ -668,6 +668,8 @@ static int ipc_pm_gate(uint32_t header)
 
 	IPC_COPY_CMD(pm_gate, ipc_get()->comp_data);
 
+	tr_info(&ipc_tr, "ipc: pm gate flags 0x%X", pm_gate.flags);
+
```
Comment (dbaluta) [inline src/ipc/handler.c:672]: What's the difference between %x and %X (lower case vs upper case)? I've mostly seen usages with lower case for x.

Comment (ktrzcinx) [reply src/ipc/handler.c:672]: 0x%x -> 0x12ab vs 0x%X 0x12AB

Comment (dbaluta) [reply src/ipc/handler.c:672]: @ktrzcinx thanks I didn't know about that :).

**Comment by dbaluta** [comment]
@ktrzcinx care to look for why the build is failing for testbench?

```
/usr/bin/cc    CMakeFiles/testbench.dir/testbench.c.o CMakeFiles/testbench.dir/alloc.c.o CMakeFiles/testbench.dir/common_test.c.o CMakeFiles/testbench.dir/file.c.o CMakeFiles/testbench.dir/ipc.c.o CMakeFiles/testbench.dir/schedule.c.o CMakeFiles/testbench.dir/ll_schedule.c.o CMakeFiles/testbench.dir/edf_schedule.c.o CMakeFiles/testbench.dir/panic.c.o CMakeFiles/testbench.dir/timer.c.o CMakeFiles/testbench.dir/topology.c.o CMakeFiles/testbench.dir/trace.c.o  -o testbench -Wl,-rpath,/home/sof/work/sof.git/tools/testbench/build_testbench/sof_ep/install/lib:/home/sof/work/sof.git/tools/testbench/build_testbench/sof_parser/install/lib: -ldl -lm sof_ep/install/lib/libsof.so sof_parser/install/lib/libsof_tplg_parser.so 

sof_ep/install/lib/libsof.so: undefined reference to `clock_get_freq'

collect2: error: ld returned 1 exit status
```

File: src/ipc/handler.c (hunk 1 above)
Comment (lgirdwood) [reply src/ipc/handler.c:672]: can we keep consistency and use the same HEX case throughout the FW. I think it's all lower case today.

File: src/lib/clk.c [hunk 2]
```diff
@@ -63,6 +65,9 @@ void clock_set_freq(int clock, uint32_t hz)
 					 hz);
 	clk_notify_data.freq = clk_info->freqs[idx].freq;
 
+	tr_info(&power_tr, "clock %d set freq %dkHz, freq_idx= %d",
```
Comment (mmaka1) [inline src/lib/clk.c:68]: '=' is used inconsistently in the log message, pls consider removing it and just keep a single space between the name and value.

File: src/audio/pipeline.c [hunk 3]
```diff
@@ -222,7 +222,8 @@ int pipeline_complete(struct pipeline *p, struct comp_dev *source,
 		.comp_data = &data,
 	};
 
-	pipe_info(p, "pipeline complete");
+	pipe_info(p, "pipeline complete, clock freq= %dkHz",
```
Comment (mmaka1) [inline src/audio/pipeline.c:225]: Recent log messages just insert single space between name and value, no extra '='.

Comment (ktrzcinx) [reply src/audio/pipeline.c]: updated

File: src/lib/clk.c (hunk 2 above)
Comment (ktrzcinx) [reply src/lib/clk.c]: updated

File: src/lib/clk.c [hunk 4]
```diff
@@ -63,6 +70,9 @@ void clock_set_freq(int clock, uint32_t hz)
 					 hz);
 	clk_notify_data.freq = clk_info->freqs[idx].freq;
 
+	tr_info(&clock_tr, "clock %d set freq %dkHz freq_idx %d",
+		clock, hz / 1000, idx);
```
Comment (lgirdwood) [inline src/lib/clk.c:74]: xtensa divison is slow. Lets send Hz and have logger do any conversions (if needed). i.e. we send all our trace frequencies in Hz and have logger do all conversions. The logger conversion can come later though in another PR.

Comment (ktrzcinx) [reply src/lib/clk.c]: updated - conversion was done only to limit number of `0` in output log, I think that making hack's in logger for such a thing is not needed, it's not important enough.

**Comment by lgirdwood** [comment]
@xiulipan @fredoh9 btw - jenkins reports seem to be missing.

**Comment by fredoh9** [comment]
> @xiulipan @fredoh9 btw - jenkins reports seem to be missing.

I checked on Jenkins, there is no task for this PR in last two days. Something wrong with Jenkins - Github link.

**Pull request merged**